commit addfa653cf51fa45c7eae49be1fd722aaadf97d6
Author: Chris Wilson <chris@chris-wilson.co.uk>
Date:   Wed Sep 26 08:59:18 2012 +0100

    uxa: Prevent attempting to call intel_mode_fini() before PreInit runs
    
    In the Ubuntu xserver it is apparently possible for the Screen to be
    deleted prior to being fully initialised. Make sure we don't crash in
    that situation!
    
    Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=55346
    Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>

diff --git a/src/intel_display.c b/src/intel_display.c
index 2a3a393..60af415 100644
--- a/src/intel_display.c
+++ b/src/intel_display.c
@@ -1770,6 +1770,9 @@ intel_mode_fini(intel_screen_private *intel)
 {
 	struct intel_mode *mode = intel->modes;
 
+	if (mode == NULL)
+		return;
+
 	while(!list_is_empty(&mode->crtcs)) {
 		xf86CrtcDestroy(list_first_entry(&mode->crtcs,
 						 struct intel_crtc,
